### DISCUSSION TITLE
PThread - signal not supported, so, how can we know when a Thread finish its work

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#16803] [Q&A]
According to documentation, PThread  signal is not supported.
so, is there a way to get an event about Thread that finish its works ?

**Comment by sbc100** [comment]
You can use shared memory to communicate between threads, either by using atomic memory accesses or via pthread APIs such as pthread_mutex and pthread_cont. 

Perhaps you are mixing up POSIX signals (which are not well supported) with pthread_cond_signal() (which is well supported?)

**Comment by aharondavid** [comment]
Ok, I will investigate that...
Thank you very match.
